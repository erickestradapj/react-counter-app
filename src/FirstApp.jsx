import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {
   return (
      <>
         <h1 data-testid="test-title"> {title} </h1>
         {/* <code>{JSON.stringify(newMessage)}</code> */}
         <p>{subtitle}</p>
         <p>{subtitle}</p>
         <p>{name}</p>
      </>
   );
};

FirstApp.propTypes = {
   title: PropTypes.string.isRequired,
   subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
   title: 'No title',
   subtitle: 'No subtitle',
   name: 'Erick E.',
};
