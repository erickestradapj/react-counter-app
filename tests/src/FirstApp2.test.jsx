import { render, screen } from '@testing-library/react';
import { FirstApp } from '../../src/FirstApp';

describe('Test in <FirstApp />', () => {
   const title = 'Hi, I am Goku';
   const subtitle = 'I am subtitle';

   test('should match with the snapshot', () => {
      const { container } = render(<FirstApp title={title} />);
      expect(container).toMatchSnapshot();
   });

   test('should show message "Hi, I am Goku"', () => {
      render(<FirstApp title={title} />);
      expect(screen.getByText(title)).toBeTruthy();
   });

   test('should show title in h1', () => {
      render(<FirstApp title={title} />);
      expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
   });

   test('should show subtitle sent by props', () => {
      render(<FirstApp title={title} subtitle={subtitle} />);
      expect(screen.getAllByText(subtitle).length).toBe(2);
   });
});
