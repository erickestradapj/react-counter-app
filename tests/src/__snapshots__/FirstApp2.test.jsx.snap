// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hi, I am Goku
     
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    No subtitle
  </p>
  <p>
    Erick E.
  </p>
</div>
`;
